src/main/java/application/AbstractBaseController.java

```diff
@@ -253,18 +253,22 @@ protected EntityDetailListResponse getInlineDetail(MenuSession menuSession) {
         if (detailId == null) {
             return null;
         }
-
         Detail persistentDetail = session.getDetail(detailId);
+        TreeReference reference = getReference(session, entityDatum);
 
-        EvaluationContext ec = session.getEvaluationContext();
-
-        TreeReference ref = entityDatum.getEntityFromID(ec, stepToFrame.getValue());
-        if (ref == null) {
-            return null;
+        EvaluationContext ec;
+        if (inline) {
+            ec = session.getEvaluationContext();
+            return new EntityDetailListResponse(persistentDetail.getFlattenedDetails(),
```

Comment:
@ctsims one question I had was about this `getFlattenedDetails()` method. This isn't used in the Android or Formplayer repositories, but seems like the correct function call here - that is, if we have a list of detail tabs we want to display those, if not we just want to use the fields in this detail. Is that assumption correct?